classes/submission/genre/Genre.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/**
+ * @file classes/submission/genre/Genre.php
+ *
+ * Copyright (c) 2014-2014 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class \PKP\genre\Genre
+ *
+ * @brief Eloquent Model for Genre
+ */
+
+
+namespace PKP\submission\genre;
+
+use Illuminate\Database\Eloquent\Builder as EloquentBuilder;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use PKP\core\traits\ModelWithSettings;
+use PKP\submissionFile\SubmissionFile;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use PKP\services\PKPSchemaService;
+use PKP\db\XMLDAO;
+
+class Genre extends Model
+{
+    use ModelWithSettings;
+
+    public const GENRE_CATEGORY_DOCUMENT = 1;
+    public const GENRE_CATEGORY_ARTWORK = 2;
+    public const GENRE_CATEGORY_SUPPLEMENTARY = 3;
+
+    protected $table = 'genres';
+    protected $primaryKey = 'genre_id';
+    public    $timestamps = false;
+    protected $guarded = ['genreId','id'];
+
+    public function getSettingsTable(): string
+    {
+        return 'genre_settings';
+    }
+
+    public static function getSchemaName(): ?string
+    {
+        return PKPSchemaService::SCHEMA_GENRE;
+    }
+
+    public function submissionFiles(): HasMany
+    {
+        return $this->hasMany(SubmissionFile::class, 'genre_id', 'genre_id');
+    }
+
+
+    // Accessors and Mutators
+    protected function entryKey(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value) => strtoupper($value),
+            set: fn($value) => strtolower($value)
+        );
+    }
+
+    protected function enabled(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value) => (bool) $value,
+            set: fn($value) => (int) $value
+        );
+    }
+
+    // Scopes
+
+    public function scopeEnabled(EloquentBuilder $query): EloquentBuilder
```

Comment:
It looks that our usual pattern is using `scope` key prefix together with `with`, i.e. `scopeWithEnabled`. For boolean values I see several cases with prefix `scopeIs...`. Anyway, it's better to use such prefix to easily identify the `scope`-related method and I'd use the former option.